packages/mui-material-next/src/ButtonGroup/ButtonGroup.tsx

```diff
@@ -137,8 +152,8 @@ const ButtonGroupRoot = styled('div', {
     ...(ownerState.variant === 'text' &&
       ownerState.color !== 'inherit' && {
         borderColor: theme.vars
-          ? `rgba(${theme.vars.palette[ownerState.color].mainChannel} / 0.5)`
-          : alpha(theme.palette[ownerState.color].main, 0.5),
+          ? `rgba(${theme.vars.palette[ownerState.color ?? 'primary'].mainChannel} / 0.5)`
```

Comment:
One thought seeing this code. Should we type the ownerState instead of extending props + additional state to have required the attributes for which we have default value, like color, size etc. In this case we are changing the runtime to fix a type issue. cc @DiegoAndai what do you think?

Reply:
@mnajdova I actually think this is a good idea, as it would also avoid some code duplication, since the nullish operator is used several times, for example. It also makes logical sense, as there is always an internal value for these values and the change of a default value would only have to take place in one place.

Reply:
Cool! Would you like to make this change?